pkg/ebpf/goruntime/goruntime.go

```diff
@@ -73,10 +76,22 @@ func (p *Tracer) SocketFilters() []*ebpf.Program {
 	return nil
 }
 
-func (p *Tracer) Run(ctx context.Context, eventsChan chan<- []ebpfcommon.HTTPRequestTrace) {
+func (p *Tracer) Run(ctx context.Context, eventsChan chan<- []interface{}) {
 	logger := slog.With("component", "goruntime.Tracer")
 	ebpfcommon.ForwardRingbuf(
-		p.Cfg, logger, p.bpfObjects.Events, nil,
+		p.Cfg, logger, p.bpfObjects.Events, p.toRequestTrace,
 		append(p.closers, &p.bpfObjects)...,
 	)(ctx, eventsChan)
 }
+
+func (p *Tracer) toRequestTrace(record *ringbuf.Record) (interface{}, error) {
```

Comment:
I'm not sure if there's elegant way we can common this code into a method we can share for all files that use the HTTPRequestTrace as struct.

Reply:
actually this doesn't need to be a method, as we don't use the `p` target variable. It could go to a common package.

Probably you can even generify it: so you can reuse the same code from other functions that do the same:

```go
func Read[T any](record *ringbuf.Record) (T, error) {
	var event T

	err := binary.Read(bytes.NewBuffer(record.RawSample), binary.LittleEndian, &event)
	if err != nil {
		slog.Error("Error reading generic HTTP event", err)
		return nil, err
	}

	return event, nil
}
```